src/components/product/product.service.ts

```diff
@@ -63,63 +74,116 @@ export class ProductService {
   }
 
   async readOne(id: string, session: ISession): Promise<Product> {
-    const result = await this.db.readProperties({
-      session,
-      id,
-      nodevar: 'product',
-      props: [
-        'id',
-        'createdAt',
-        'type',
-        'books',
-        'mediums',
-        'purposes',
-        'approach',
-        'methodology',
-      ],
-    });
+    const readProduct = this.db
+      .query()
+      .match(matchSession(session, { withAclRead: 'canReadProducts' }))
+      .match([
+        node('product', 'Product', {
+          active: true,
+          id: id,
+        }),
+      ])
+      .optionalMatch([...this.propMatch('type')])
+      .optionalMatch([...this.propMatch('books')])
+      .optionalMatch([...this.propMatch('mediums')])
+      .optionalMatch([...this.propMatch('purposes')])
+      .optionalMatch([...this.propMatch('approach')])
+      .optionalMatch([...this.propMatch('methodology')])
+      .return({
+        product: [{ id: 'id' }],
+        type: [{ value: 'type' }],
+        books: [{ value: 'books' }],
+        mediums: [{ value: 'mediums' }],
+        purposes: [{ value: 'purposes' }],
+        approach: [{ value: 'approach' }],
+        methodology: [{ value: 'methodology' }],
+      });
+
+    const result = await readProduct.first();
 
-    if (!result || !result.id.value) {
+    if (!result || !result.id) {
+      this.logger.warning(`Could not find product`, { id: id });
       throw new NotFoundException('Could not find product');
     }
 
     return {
       id,
-      createdAt: result.createdAt.value,
-      type: result.type.value,
-      books: result.books?.value || [],
-      mediums: result.mediums?.value || [],
-      purposes: result.purposes?.value || [],
-      approach: result.approach.value,
-      methodology: result.methodology.value,
+      createdAt: result.createdAt,
+      type: result.type,
+      books: result.books,
+      mediums: result.mediums,
+      purposes: result.purposes,
+      approach: result.approach,
+      methodology: result.methodology,
     };
```

Comment:
This needs to be in the `query.helpers.ts` style. Use the `organization` service's `readOne` function as an example.